### PR TITLE
For #23144 Address regression in COLD start

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -397,7 +398,11 @@ class HomeFragment : Fragment() {
 
         if (shouldEnableWallpaper()) {
             val wallpaperManger = requireComponents.wallpaperManager
-            wallpaperManger.updateWallpaper(binding.homeLayout, wallpaperManger.currentWallpaper)
+            // We only want to update the wallpaper when it's different from the default one
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:401]: 🤦🏽‍♂️ moving it. Thanks for caching this!

**Review by csadilek — APPROVED** [review]
👍

File: app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt [hunk 2]
```diff
@@ -15,7 +16,7 @@ import org.mozilla.fenix.utils.Settings
  * Provides access to available wallpapers and manages their states.
  */
 class WallpaperManager(private val settings: Settings) {
-
+    val logger = Logger("WallpaperManager")
```
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt:19]: Are these logs published somewhere or are they just for debugging?

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt:19]: They go to the logcat console, it's just for debugging :)

**Pull request merged**